GoT/app/controllers/characters_controller.rb

```diff
@@ -0,0 +1,57 @@
+class CharactersController < ApplicationController
+
+
+  # index
+def index
+  @character = Character.all
+end
+
+# new
+def new
+  @house = House.find(params[:house_id])
+  @character = Character.new
+end
+
+# show
+def show
+@character = Character.find(params[:house_id])
+end
+
+
+   def create
+     @house = House.find(params[:house_id])
+     @character = @house.characters.create(character_params)
+     redirect_to "/houses/#{@house.id}"
+   end
+
+
+   def edit
+      @house = House.find(params[:house_id])
+      @character = Character.find(params[:id])
+   end
+
+   # update
+     def update
+       @house = House.find(params[:house_id])
+       @character = @house.characters.find(params[:id])
+        @character.update(character_params)
+
+       redirect_to house_path(@house)
+     end
+
+
+   def destroy
+      @house = House.find(params[:house_id])
+      @character = Character.find(params[:id])
+      @character.destroy
+
+    redirect_to "/houses"
+
+      end
+
+    private
+   def character_params
+     params.require(:character).permit(:name, :photo_url, :title, :quotes, :house_id)
```

Comment:
Thanks,
Ya when I am pulling the house_id from the path that's all I need to perform all RESTful functionalities. interns of "White listing" or "Black listing" a strong param I didn't need to add house_id. if I were to let users edit the house_id to move one character from one house to another by just putting the house_id I could have done that. Just a waste of resources. I would actually like to talk to you about this matter if you know where I can look it up when you have a big project in which you have to worry about what is permitted or not.